### DISCUSSION TITLE
Best Practices to use property as external entity

### DISCUSSION THREAD

**Discussion by radaniba** [marklogic-community/envision#261] [Q&A] [answered]
Hi @wpaven 
I have a very general question that I asked [here](https://stackoverflow.com/questions/67666193/marklogic-entity-property-versus-relationship-for-an-effective-mastering) and I will share some updates when I got some, please feel free to comment as well.

But my question here is different, it is regarding DHCCE.

When I use an entity property with a type (related entity) , after we do the matching and merging, the related entites don't show up on the graph (network) 

But when I follow your example and do relation between entities they do show up, and I am wondering here what is the best practices 

Can I please ask you to share the JSON document of HR360 example ? I would like to see how the entities showup on a single json document when you do the matching merging

Rad

**Comment by wpaven** [comment, marked as answer]
So, relationships in MarkLogic for entities today are more like relational type relationships.  Behind the scenes, a tabular view for each entity is created by default.  
Let's say you create a Customer entity.  Behind the scenes a Customer.entitity.json entity descriptor is created.  A TDE for each entity is auto-generated by the hub based on this and if you go to query console after harmonization (assuming user has TDE-view, TDE-admin role) you can query the Customer "table".  
When you use properties to define the relationship inside the entity descriptor, this is more like a relationship between the "tables". It's not a named relationship, as you're seeing in the graph view in Connect.
In Connect. we augment the auto-generated TDE with another TDE that has the named relationships. This is what is showing up in Explore. 
This is how Connect works because entity services doesn't support named relationships in the descriptor, today.
However, the plan is to update entity services in an upcoming Data Hub release and at that time, then if you look at the Customer.entity.json, you'd see named relationships between entities defined. But today, again, you won't find those name relationships in the descriptor.  So, that's some of the secret sauce of Connect and Explore. In DHCCE we augment the the auto-generated TDE with an additional one for named relationships.

Hope this helps,
Pete

**Reply by radaniba** [reply]
Thank you @wpaven as usual very very helpful

**Reply by radaniba** [reply]
@wpaven  would you please be able to share an sample of hr360 example with a mastered - merged entity ?
curious to see how this looks like in the named relationship once merged to plan postprocessing of data searching eventually
Rad

**Reply by wpaven** [reply]
We've actually made the HR350 use case available for free on Docker Hub.
https://hub.docker.com/_/marklogic-datahub-hr360
A single image contains MarkLogic Server, Data Hub Central Community Edition, Hub Central, Pipes, and the HR360 use case all pre-configured and ready to use.  You just pull the image, spin it up, and you're in business.  You can login to Query Console to examine everything in the databases.
https://marklogic-community.github.io/data-hub-central-community/docker/
Does this help?

**Reply by radaniba** [reply]
It sure does !! Thanks a lot @wpaven
